### ISSUE TITLE
NPE in ServerRequestCreateUrl.updateShareEventToFabric

### ISSUE THREAD

**Issue by jakub-g** [BranchMetrics/android-branch-deep-linking#498]
Hello,

I've noticed this NPE which was happening a lot for me on Friday (not happening ever before, not sure what is the trigger) and prevented the share with branch from working properly in our app.

We use 2.11.1 so we are a few commits behind, but looking at the diff, updating to latest probably won't fix the bug (I'll try though).

This the problematic line

https://github.com/BranchMetrics/android-branch-deep-linking/blob/master/Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java#L261

AFAICS, there are two constructors, one is initializing `linkPost_` but another is not, and hence probably the nullpointer?

Stacktrace:

```
11-03 09:26:45.174 30794-30794/? W/System.err: java.lang.NullPointerException: Attempt to invoke virtual method 'org.json.JSONObject io.branch.referral.BranchLinkData.getLinkDataJsonObject()' on a null object reference
11-03 09:26:45.174 30794-30794/? W/System.err:     at io.branch.referral.ServerRequestCreateUrl.updateShareEventToFabric(ServerRequestCreateUrl.java:255)
11-03 09:26:45.174 30794-30794/? W/System.err:     at io.branch.referral.ServerRequestCreateUrl.onRequestSucceeded(ServerRequestCreateUrl.java:126)
11-03 09:26:45.174 30794-30794/? W/System.err:     at io.branch.referral.Branch$BranchPostTask.onPostExecute(Branch.java:2767)
11-03 09:26:45.174 30794-30794/? W/System.err:     at io.branch.referral.Branch$BranchPostTask.onPostExecute(Branch.java:2609)
```

**Comment by sojanpr** [comment]
@jakub-g Sorry for this issue. We have added fix in this PR https://github.com/BranchMetrics/android-branch-deep-linking/issues/498

Basically above issue can occur only when  link creation is failed due to some error. We persist the request and resend it in some cases. 
Note:  can you please check if app by any chance creating  a Branch link before   Branch SDK is initialized (i.e., before Branch#initSession() is called)?

**Comment by sojanpr** [comment]
Test Steps:
1. Launch the testbed.
2. call `BUO#getShortUrl()` before init happens and kill the app before init
3. Open the app and see  if any pending `v1/url` request are send to the server.
4. Make sure testbed app is not crashing in step 4
5. Make sure link creation event is reported to Fabric

**Comment by sojanpr** [comment]
Fix added to v2.14.0
